File: src/legalintel/models/docket.py
from datetime import date, datetime

from pydantic import BaseModel


class TrackDocketRequest(BaseModel):
    courtlistener_docket_id: int
    matter_id: int | None = None


class TrackedDocket(BaseModel):
    id: int
    courtlistener_docket_id: int
    court: str | None
    docket_number: str | None
    case_name: str | None
    matter_id: int | None
    created_at: datetime
    last_checked_at: datetime | None


class DocketEntry(BaseModel):
    courtlistener_entry_id: int
    entry_number: int | None
    description: str
    date_filed: date | None


class DocketAlert(BaseModel):
    id: int
    tracked_docket_id: int
    created_at: datetime
    new_entry_count: int
    new_entry_ids: list[int]


class DocketCheckResult(BaseModel):
    tracked_docket_id: int
    checked_at: datetime
    new_entries: list[DocketEntry]
    alert_created: bool
    alert: DocketAlert | None = None

File: frontend/src/types/docket.ts
// Mirrors src/legalintel/models/docket.py 1:1.

export interface TrackedDocket {
  id: number;
  courtlistener_docket_id: number;
  court: string | null;
  docket_number: string | null;
  case_name: string | null;
  matter_id: number | null;
  created_at: string;
  last_checked_at: string | null;
}

export interface DocketEntry {
  courtlistener_entry_id: number;
  entry_number: number | null;
  description: string;
  date_filed: string | null;
}

export interface DocketAlert {
  id: number;
  tracked_docket_id: number;
  created_at: string;
  new_entry_count: number;
  new_entry_ids: number[];
}

export interface DocketCheckResult {
  tracked_docket_id: number;
  checked_at: string;
  new_entries: DocketEntry[];
  alert_created: boolean;
  alert: DocketAlert | null;
}
